shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/update/DropShardingAlgorithmStatementUpdater.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.shardingsphere.infra.metadata.resource.ShardingSphereResource;
+import org.apache.shardingsphere.sharding.api.config.ShardingRuleConfiguration;
+import org.apache.shardingsphere.sharding.distsql.parser.statement.DropShardingAlgorithmStatement;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.stream.Collectors;
+
+/**
+ * Drop sharding algorithm statement updater.
+ */
+public final class DropShardingAlgorithmStatementUpdater implements RuleDefinitionDropUpdater<DropShardingAlgorithmStatement, ShardingRuleConfiguration> {
+    
+    @Override
+    public void checkSQLStatement(final String schemaName, final DropShardingAlgorithmStatement sqlStatement, 
+                                  final ShardingRuleConfiguration currentRuleConfig, final ShardingSphereResource resource) throws RuleDefinitionViolationException {
+        checkCurrentRuleConfiguration(schemaName, currentRuleConfig);
+        checkShardingAlgorithmsExisted(schemaName, sqlStatement, currentRuleConfig);
+        checkShardingAlgorithmsInUsed(schemaName, sqlStatement, currentRuleConfig);
+    }
+    
+    private void checkCurrentRuleConfiguration(final String schemaName, final ShardingRuleConfiguration currentRuleConfig) throws RequiredAlgorithmMissedException {
+        if (null == currentRuleConfig) {
+            throw new RequiredAlgorithmMissedException(schemaName);
+        }
+    }
+    
+    private void checkShardingAlgorithmsExisted(final String schemaName, final DropShardingAlgorithmStatement sqlStatement,
+                                                final ShardingRuleConfiguration currentRuleConfig) throws RequiredAlgorithmMissedException {
+        Collection<String> currentShardingAlgorithms = getCurrentShardingAlgorithms(currentRuleConfig);
+        Collection<String> notExistedAlgorithms = sqlStatement.getAlgorithmNames().stream().filter(each -> !currentShardingAlgorithms.contains(each)).collect(Collectors.toList());
+        if (!notExistedAlgorithms.isEmpty()) {
+            throw new RequiredAlgorithmMissedException(schemaName, notExistedAlgorithms);
+        }
+    }
+    
+    private void checkShardingAlgorithmsInUsed(final String schemaName, final DropShardingAlgorithmStatement sqlStatement, final ShardingRuleConfiguration currentRuleConfig) 
+            throws AlgorithmInUsedException {
+        Collection<String> allInUsed = getAllOfAlgorithmsInUsed(currentRuleConfig);
+        Collection<String> usedAlgorithms = sqlStatement.getAlgorithmNames().stream().filter(allInUsed::contains).collect(Collectors.toList());
+        if (!usedAlgorithms.isEmpty()) {
+            throw new AlgorithmInUsedException(schemaName, usedAlgorithms);
+        }
+    }
+    
+    private Collection<String> getAllOfAlgorithmsInUsed(final ShardingRuleConfiguration shardingRuleConfig) {
+        Collection<String> result = new LinkedHashSet<>();
+        shardingRuleConfig.getTables().stream().forEach(each -> {
```

Comment:
autoTables in ShardingRuleConfiguration should also be added

Reply:
Good idea 👍